Sources/XCMetricsBackendLib/UploadMetrics/Repository/AzureRepository/LogFileABSRepository.swift

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2020 Spotify AB.
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import Foundation
+import Vapor
+
+/// `LogFileRepository` that uses Azure Blob Storage to store and fetch logs
+struct LogFileABSRepository: LogFileRepository {
+    let storageAccountName: String
+    let storageContainerName: String
+    let storageAccountAccessKey: String
+    
+    let client: Client
+
```

Comment:
should follow same pattern as [LogFileS3Repository](https://github.com/tomtom-forks/XCMetrics/blob/main/Sources/XCMetricsBackendLib/UploadMetrics/Repository/LogFileS3Repository.swift) or [LogFileGCSRepository](https://github.com/tomtom-forks/XCMetrics/blob/main/Sources/XCMetricsBackendLib/UploadMetrics/Repository/LogFileGCSRepository.swift)